<?xml version="1.0"?>
<launch>
    <!-- Gazebo options -->
    <arg name="ns" default="/" />
    <arg name="use_rviz" default="true" />
    <arg name="benchmark" default="false" />
    <arg name="random_shortcut" default="true" />
    <arg name="random_shortcut_time" default="0.1" />
    <arg name="tight_shortcut" default="true" />
    <arg name="output_file" default="$(find mr_planner)/outputs/dual_gp4_benchmark.csv" />
    <arg name="planner_name" default="RRTConnect" />
    <arg name="planning_time_limit" default="2.0" />

    <group ns="$(arg ns)">
        <!-- Start the robot state publisher, simulation, and controler -->
        <!-- <include file="$(find dual_gp4_moveit_config)/launch/demo_gazebo.launch">
        </include> -->

        <include file="$(find panda_three_moveit_config)/launch/demo.launch">
            <arg name="use_rviz" value="$(arg use_rviz)" />
        </include>

        <!-- Start the demo test node -->
        <node name="test_collision" pkg="mr_planner" type="demo" output="screen" required="true">
            <param name="num_robots" value="3" />
            <param name="movegroup_name" value="panda_multi_arm" />
            <param name="group_name_0" value="panda0_arm" />
            <param name="group_name_1" value="panda1_arm" />
            <param name="group_name_2" value="panda2_arm" />
            <param name="mfi" value="false" />
            <param name="async" value="false" />
            <param name="shortcut" value="true" />
            <param name="benchmark" value="$(arg benchmark)" />
            <param name="random_shortcut" value="$(arg random_shortcut)" />
            <param name="random_shortcut_time" value="$(arg random_shortcut_time)" />
            <param name="tight_shortcut" value="$(arg tight_shortcut)" />
            <param name="planner_name" value="$(arg planner_name)" />
            <param name="planning_time_limit" value="$(arg planning_time_limit)" />
            <param name="benchmark" value="$(arg benchmark)" />
            <param name="output_file" value="$(arg output_file)" />

            <param name="pose_name" value="wall1_pose" />
            <param name="pose_name1" value="shelf_0" />
            <param name="pose_name2" value="shelf_1" />
            <param name="pose_name3" value="shelf_2" />
            <param name="pose_name4" value="shelf_3" />
            <param name="pose_name5" value="shelf_multi_0" />
            <param name="pose_name6" value="shelf_multi_1" />
            <param name="pose_name7" value="shelf_multi_2" />
            <param name="pose_name8" value="wall_0" />
            <param name="pose_name9" value="wall_1" />
            <param name="pose_name10" value="wall_2" />
        </node>
    </group>
</launch>